### DISCUSSION TITLE
What is the output segmenting for ListenAsync text events?

### DISCUSSION THREAD

**Discussion by paviad** [Tyrrrz/CliWrap#180] [General]
When `ListenAsync` emits a standard output event, is it guaranteed to be a full line?

What if the process outputs a progress dot `.` every second, would there be an event for each dot? Or only when a new line is output?

Thanks

**Comment by Tyrrrz** [comment]
It yields new input on line breaks, so yeah, guaranteed to be a full line.

Also note that even if you write code like this in the underlying app:

```csharp
Console.Write("foo");
Console.Write("bar");
```

In reality, this will write the data to an internal buffer and only flush the data to stdout once it reaches a certain size (or if you do `Console.Out.Flush()`). From CliWrap's side, we can't reliably tell when the process actually issues those writes -- we're just working with a contiguous stream of data.

**Reply by paviad** [reply]
Thank you very much for the quick reply!

This is an important part of the API contract. You might want to state it explicitly in your API documentation (referring to the fact that only a line break will yield a new event in `ListenAsync`)

**Reply by Tyrrrz** [reply]
It does already though, doesn't it? Or do you find "new line" ambiguous in this context?

![image](https://user-images.githubusercontent.com/1935960/204248534-3da52cb4-1483-4f5e-8e23-c744f91df8c0.png)

**Comment by paviad** [comment]
Indeed it does! I must have missed that, anyway, good job with the library, **extremely** useful.
